base/strings/types.jl

```diff
@@ -100,8 +100,41 @@ function thisind(s::SubString{String}, i::Integer)
     j-offset
 end
 
-nextind(s::SubString, i::Integer) = nextind(s.string, i+s.offset)-s.offset
-prevind(s::SubString, i::Integer) = prevind(s.string, i+s.offset)-s.offset
+# need to define nextind and prevind only for SubString{String}
+# as other cases are handled by definitions for AbstractString
```

Comment:
These generic definitions for `SubString{T}` should be more efficient than the generic `nextind` for `AbstractString` if there is an efficient `nextind` for `T`.  So, better to leave them in.

Reply:
Oh never mind, I guess we already discussed that above?:

Reply:
The PR is outdated (it was prepared before "string overhaul"), but the problem it solved is still present in the new code.

Therefore my thinking is to first decide on the desired behavior (I guess @StefanKarpinski will have an opinion here).

Next there is an issue of maximum efficiency of implementation. My experience is that at least `SubString{String}` should have a specialized method - as this will be most common case in practice and probably there is a room for efficiency improvement by using custom code.

But first we should have a defined contract we want the implementation to follow.